packages/core/src/api/nodeUtil.ts

```diff
@@ -264,8 +267,9 @@ export function getBlocksChangedByTransaction<
       const prev = prevBlocks[id];
       const next = nextBlocks[id];
       const isParentDifferent = prev.parentId !== next.parentId;
+      const isIndexDifferent = prev.index !== next.index;
```

Comment:
Hey @nperez0111 👋 

This solution tracks the previous index and registers a change if it changes. This results in every affected block being included in the changes (not just the one that was dragged), see the attached video for reference.

LMK if you like this approach. If it's good, I'll update the snapshots.

https://github.com/user-attachments/assets/34d47545-210a-49b0-9e39-9d3877ab6906

Reply:
Yeah it's too much noise. I'm thinking at most we'd include the moved blocks plus any blocks that might've a new parent. Detecting the moved block with just indexes won't do it though because when rotating arrays we might detect multiple. Any ideas? I'm looking to see if it's possible to mark them as "dragged" by the time we determine changes.